### ISSUE TITLE
Provide 'validate' option

### ISSUE THREAD

**Issue by kampka** [nix-community/nixpkgs-fmt#147]
**Is your feature request related to a problem? Please describe.**
I usually use formaters / linters in pre-commit hooks to prevent mal-formatted commits but I dislike for the formatter to manipulate the git index by formatting changed files. Instead, I prefer running in linter mode that can fail the hook with exit code != 0 in case of formatting issues. The point of this is that I prefer to review the change before committing it. `nixpkgs-fmt` lacks such a mode.

**Describe the solution you'd like**
A command option `nixpkgs-fmt --validate [srcs]` that takes a list of files and fails if one or more files are not properly formatted.

**Describe alternatives you've considered**
I suppose you can achieve this using bash scripting, eg (pseudo-code)
```bash
for file in $(find . -name "*.nix"); do
  cat "$file" | nixpkgs-fmt --explain >  "$file.explain"
  if diff "$file" "$file.explain"; then
    exit 1
  fi 
done
exit 0
```

This does feel like a hack though.

**Comment by zimbatm** [comment]
Yep, it would be nice to get that feature. Ideally the output would be formatted in the diff format as well.

**Comment by kampka** [comment]
I think it would be nice if the output was less verbose or maybe has a switch, eg. combining validate and explain.

**Comment by Mic92** [comment]
In [black](https://github.com/psf/black) the UX looks like this:

```
$ black --check nix_review
All done! ✨ 🍰 ✨
19 files would be left unchanged.
$ echo $?
0
$ black --check nix_review
would reformat /home/joerg/git/nix-review/nix_review/review.py
All done! 💥 💔 💥
1 file would be reformatted, 18 files would be left unchanged.
$ echo $?
1
```

**Comment by zimbatm** [comment]
Now available in nixpkgs-fmt 0.6.0
